lib/matplotlib/contour.py

```diff
@@ -969,19 +969,29 @@ def _autolev(self, N):
         Select contour levels to span the data.
 
         The target number of levels, *N*, is used only when the
-        scale is not log and default locator is used.
+        locator is not set and the scale is log or the default
+        locator is used.
```

Comment:
> If *N* is an int, it is used as the target number of levels.
> Otherwise (N = None), a reasonable default is chosen;
> For logscales the LogLocators default, N=7 otherwise.